index.d.ts

```diff
@@ -0,0 +1,60 @@
+export interface Action {
+  type: string;
+}
+
+export type Reducer<S> = (state: S, action: Action) => S;
+
+export type Dispatch = (action: any) => any;
```

Comment:
We can. Just make it generic:

```
interface Dispatch extends Function {
    <A>(action: A): A;
}
```

or if dispatch returns output changed by middleware:
```
interface Dispatch extends Function {
    <A, O>(action: A): O;
}
```

This is one of typings that is necessary for strict typing of any redux based app. If dispatch can return anything and can't be parametrised then static typing is of no value here…